packages/flame/lib/src/effects2/destroy_effect.dart

```diff
@@ -0,0 +1,17 @@
+
+import 'effect.dart';
+import 'simple_effect_controller.dart';
+
+/// This simple effect, when attached to a component will cause that component
+/// to be removed from the game tree and destroyed after `delay` seconds.
+class DestroyEffect extends Effect {
```

Comment:
I would name this RemoveEffect to match the shoulRemove attrition, or the removeFromParent method

Reply:
sounds good